### ISSUE TITLE
Support non-SemVer and en dashes

### ISSUE THREAD

**Issue by koppor** [marcaddeo/clparse#22]
When trying to parse https://github.com/JabRef/jabref/blob/v5.11/CHANGELOG.md, `version` is always `null`. I think, it could be because of the used en dash. There is an issue, because the official format might permit en dash (see https://github.com/olivierlacan/keep-a-changelog/discussions/497; linked from https://github.com/nbbrd/heylogs/pull/140#issuecomment-1713265321).

``` json
    {
      "version": null,
      "link": "https://github.com/JabRef/jabref/compare/v5.10...v5.11",
      "date": null,
```

If en dash support is "difficult" to implement, maybe this tool coud parse the github compare link to get the version out of that link?

(General background: I was recommended clparse at https://github.com/NiclasvanEyk/keepac/issues/20#issuecomment-1712944920)

**Comment by koppor** [comment]
Update: Sometimes, software uses keep-a-changelog, but do not follow SemVer. JabRef is one example. It would be nice if this tool was usable with JabRef, too.

**Comment by marcaddeo** [comment]
Hi @koppor, those seem like reasonable requests. I'm going to add a flag (`--separator`) that you can pass the en dash with. This will use it for parsing the initial changelog as well as when outputting markdown versions.

I've also updated it to allow more permissive version strings, instead of only SemVer.

I took a look at the JabRef changelog, and saw that you guys have some very long changelog entries and went ahead and also added the ability to disable entry line wrapping and the ability to wrap at a custom character count (`--no-wrap` & `--wrap-at`).

I've got it in a PR (#23) for now, and will try and get a release out later today or tomorrow.

**Comment by marcaddeo** [comment]
One other thing I wanted to mention: The JabRef changelog has an "Older versions" section at the bottom of it, which is not really in line with Keep a Changelog and thus does not parse very well with `clparse` (it creates an additional unreleased section at the bottom).

This may still be acceptable for your use-cases, as I think you're just looking to diff things, but I'm not sure how to support that.

**Comment by marcaddeo** [comment]
@koppor I've released `clparse 0.9.0`. However, it's only currently able to be installed via `cargo` as I'm having trouble generating releases. I haven't updated this repository since before I got a new MacBook with an M1 (arm64) processor. So I'm going to have to stand up a GitHub Actions build workflow to get those back in place.

**Comment by marcaddeo** [comment]
Released 0.9.1, should be all set now.
